common/nymcoconut/src/scheme/keygen.rs

```diff
@@ -219,7 +265,7 @@ where
     {
         let mut peekable = iter.peekable();
         let head_attributes = match peekable.peek() {
-            Some(head) => head.borrow().beta.len(),
+            Some(head) => head.borrow().beta_g2.len(),
```

Comment:
If I understood it correctly, I think we might need an additional check here to make sure `beta_g2.len() == beta_g1.len()`